homeassistant/components/geniushub/binary_sensor.py

```diff
@@ -1,52 +1,43 @@
 """Support for Genius Hub binary_sensor devices."""
-from typing import Any, Dict
-
 from homeassistant.components.binary_sensor import BinarySensorDevice
-from homeassistant.util.dt import utc_from_timestamp
 
-from . import DOMAIN, GeniusEntity
+from . import DOMAIN, GeniusDevice
 
-GH_IS_SWITCH = ["Dual Channel Receiver", "Electric Switch", "Smart Plug"]
+GH_STATE_ATTR = "outputOnOff"
 
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the Genius Hub sensor entities."""
+    if discovery_info is None:
+        return
+
     client = hass.data[DOMAIN]["client"]
 
     switches = [
-        GeniusBinarySensor(d) for d in client.device_objs if d.type[:21] in GH_IS_SWITCH
+        GeniusBinarySensor(d, GH_STATE_ATTR)
+        for d in client.device_objs
+        if GH_STATE_ATTR in d.data["state"]
     ]
 
-    async_add_entities(switches)
+    async_add_entities(switches, update_before_add=True)
 
 
-class GeniusBinarySensor(GeniusEntity, BinarySensorDevice):
+class GeniusBinarySensor(GeniusDevice, BinarySensorDevice):
     """Representation of a Genius Hub binary_sensor."""
 
-    def __init__(self, device) -> None:
+    def __init__(self, device, state_attr) -> None:
         """Initialize the binary sensor."""
         super().__init__()
 
         self._device = device
+        self._state_attr = state_attr
+
         if device.type[:21] == "Dual Channel Receiver":
-            self._name = f"Dual Channel Receiver {device.id}"
+            self._name = f"{device.type[:21]} {device.id}"
```

Comment:
Can we set a unique_id?

Reply:
The Zones (Climate, WaterHeater), and Devices (Sensor, BinarySensor) have fixed IDs (ordinals, really), but they are a little problematic 
- there is no ID for the Hub (which is a collection of Zones, Devices) (so people, like me, with multiple hubs will have a problem) - I could use the Hub name...
 - removing, and re-adding a device gives it a new ID (is a problem anyway)

I haven't quite convinced myself it's the right thing to do...

There may be an ID lurking in the v3 JSON, but that causes issues with support the v1 JSON, which we should always support, as it is the 'official' API.

Reply:
I think everyone would be using v3 API, so will write code to use that, or not...

...give me a sec.

Reply:
I have added a `unique_id` for those using v3 API, and taken away `entity_id` for now.